libs/sdk-ui-filters/src/AttributeFilter/utils/useDeepEffect.ts

```diff
@@ -0,0 +1,21 @@
+// (C) 2021 GoodData Corporation
+import { useRef, useEffect } from "react";
+import isEqual from "lodash/isEqual";
+
+/**
+ * TODO this is not a correct directory to place hook like this in. We need to figure out the correct location
+ */
+export const useDeepEffect = (effectFunc: () => any, deps: React.DependencyList) => {
+    const isFirst = useRef(true);
+    const prevDeps = useRef(deps);
+    useEffect(() => {
+        const isSame = prevDeps.current.every((obj, index) => isEqual(obj, deps[index]));
+        if (isFirst.current || !isSame) {
+            effectFunc();
```

Comment:
This can be problematic, as you have no guarantee about the promise resolution order, right?

Example:
What about if `getElements()` promise in `useDeepEffect` hook is fired multiple times (e.g. 2x or more, with different deps). You have 2 or more running promises in parallel, but you are interested only about the latest one fired, which contains the correct data. What about if the first promise is resolved later than the second promise (with latest required data)?

Or are we interested about all the results, and resolution order is not important? (I'm not sure. If so, then it's likely ok, just add docs/comments to this hook and move it to @gooddata/sdk-ui)

Note: this is reason why useCancelablePromise exists - it cares only about the latest promise result.